### PR TITLE
test: add E2E tests for maximum privacy security check during onboarding

### PR DESCRIPTION
**Pull request by chloeYue**
## **Description**

1. Adds E2E tests to verify that with maximum privacy settings enabled during onboarding, only allowlisted network calls are made. Covers both import wallet and create new wallet user flows.
2. Add allowlist file `privacy-max-allowlist-onboarding.json`
3. Add code owner for allowlist

So we ensure users who enable maximum privacy during onboarding only make expected network calls. 
example log when test fails:

Screenshot 2025-12-03 at 15 34 04

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/38555?quickstart=1)

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/MMQA-1154

## **Manual testing steps**

Test pass and stable.
Test should fail if new host is called during onboarding which is not in allowlist 

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> Adds E2E tests that verify only allowlisted hosts are contacted during and after onboarding, introduces the allowlist JSON, and updates CODEOWNERS.
> 
> - **E2E Tests**:
>   - Add `test/e2e/tests/privacy/onboarding-maximum-privacy.spec.ts` to enforce maximum-privacy networking during onboarding.
>     - Covers both `importSRPOnboardingFlow` and `createNewWalletOnboardingFlow`.
>     - Mocks Infura RPC and feature flags, captures request hosts, and asserts against `duringOnboarding` and `untilOnboardingComplete` allowlists.
> - **Test Data**:
>   - Add `test/e2e/tests/privacy/privacy-max-allowlist-onboarding.json` defining allowlisted hosts for onboarding phases.
> - **Repo Ownership**:
>   - Update `.github/CODEOWNERS` to include QA and extension-privacy-reviewers for the new allowlist and E2E paths.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 163d4b7781c731c65f144963ce1f3e0b31930c0d. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: test/e2e/tests/privacy/privacy-max-allowlist-onboarding.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "duringOnboarding": ["accounts.google.com", "metamask.github.io"],
+  "toBeRemoved": {
```
Comment (chloeYue) [inline test/e2e/tests/privacy/privacy-max-allowlist-onboarding.json:8]: this `toBeRemoved`  list is a reminder, we need to remove the 2 hosts from allowlist once they are addressed

File: test/e2e/tests/privacy/privacy-max-allowlist-onboarding.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "duringOnboarding": [
+    "accounts.google.com",
+    "firefox.settings.services.mozilla.com",
```
Comment (chloeYue) [inline test/e2e/tests/privacy/privacy-max-allowlist-onboarding.json:4]: firefox browser-level call, only exists on firefox, not initiated by extension

**Review by DDDDDanica — APPROVED** [review]

**Review by seaona — APPROVED** [review]
awesome tests 🙌

**Review by Gudahtt — APPROVED** [review]
LGTM!
